### PR TITLE
Remove obsolete DEB package name checks

### PR DESCRIPTION
**Pull request by MichaIng**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
The installer contains three checks for specific DEB package names, which did change in past Debian/Ubuntu versions. These checks are obsolete with the current set of supported distro versions:

**iproute** vs **iproute2**: All distro versions down to Debian Jessie and Ubuntu Xenial ship the iproute2 package:
- https://packages.debian.org/search?keywords=iproute
- https://packages.ubuntu.com/search?suite=all&keywords=iproute

**php5** vs **php**: None of the Ubuntu version down to Xenial and only Debian Jessie ships the php5 package:
- https://packages.debian.org/search?keywords=php5
- https://packages.ubuntu.com/search?suite=all&keywords=php5
Moreover, installs with PHP5 would fail anyway for a longer time, due to the added php-xml module package, which became a dedicated package with PHP7.0 while being part of the core package with PHP5:
- https://packages.debian.org/search?keywords=php5-xml

**php-sqlite** vs **php-sqlite3**: With PHP7, the SQLite module package name changed to sqlite3 prefix:
- https://packages.debian.org/search?keywords=php-sqlite
- https://packages.ubuntu.com/search?suite=all&keywords=php-sqlite

Additionally the code comment about minimal apt-get call output was moved to the actual apt-get install call section, as if refers to issues with package installs that require interactive action and hence output about it to the console. The package cache update as well requires an interactive confirmation when the underlying suite code name changes, e.g. when "buster" becomes "oldstable" and "bullseye" becomes "stable". But that is not what the referred issue was about.

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

**Comment by PromoFaux** [comment]
Thanks, though, regards all the php stuff, one thing worth noting here:

https://github.com/pi-hole/pi-hole/blob/development-v6/automated%20install/basic-install.sh

**Comment by MichaIng** [comment]
I'll port it to the v6 branch when accepted, which is then only the `iproute2` check removal (and moving the code comment).

I was thinking if its worth to still change something on pre-v6 installer: Had this on the list for a longer time and as long as there is not ETA/beta vor Pi-hole v6, removing obsolete code can at least not hurt, so my thought, bringing both code bases closer together.

**Comment by PromoFaux** [comment]
> Or will v5 still be maintained long enough so that its worth it?

I want to say v6 will be this year, maybe even summer, but that's certainly not an ETA! It all depends on factors.

**Comment by MichaIng** [comment]
Okay, not sure whether it's worth it for the v5 development branch then, I can re-open it against development-v6, respectively simply change the base branch and resolve conflicts. Please advice.

Just to add, aside of removing obsolete code, this also solves practical errors, as `apt-cache show` is not a check for actually available APT packages, but returns "true" as well when any installed (e.g. 3rd party) package contains the name as conflict, "replaces", "provides" or any other kind of alias. This is not that uncommon for the involved packages, obviously as those name existed. When the package lists/cache update fails, e.g. because of failing DNS (wich does not make `apt-get update` return an error => installer continues), this can lead to falsely selected package names, which was the reason why I aimed to change it to the more explicit check, which was reverted [here](https://github.com/pi-hole/pi-hole/pull/3997), as that method fails as well when `apt-get update` continued with an DNS error. By removing the code, the DNS-related errors are deferred to `apt-get install` calls, which is better because it's output is not redirected, hence one can see DNS errors on console instead of facing a weird failure when PHP5 packages, or `iproute` (without `2`) fail to be installed afterwards. Adding a DNS check before any other network call with redirected output is another idea to address the source of such issues, but in a separate PR.

**Comment by PromoFaux** [comment]
I think that any reworks of this nature are going to be best off targeting v6. We will have an open beta round at some point before go-live (worked well for v5!) so plenty more change to catch bugs before they are out in the wild.

We're planning another 5.x release in the next day or so, and that _should_ be the last major one (Should be...)

**Comment by MichaIng** [comment]
Okay, makes sense then to switch this over to `development-v6`.

**Comment by MichaIng** [comment]
Waiting for #4218 (which touches some same code lines) and will then resolve conflicts here.

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -303,64 +303,25 @@ if is_command apt-get ; then
...
-    elif apt-cache show "${phpVer}-sqlite" > /dev/null 2>&1; then
-        phpSqlite="sqlite"
-    else
-        printf "  %b Aborting installation: No SQLite PHP module was found in APT repository.\\n" "${CROSS}"
-        exit 1
-    fi
-    # Packages required to run this install script (stored as an array)
-    INSTALLER_DEPS=(dhcpcd5 git "${iproute_pkg}" whiptail dnsutils)
+
+    # Since our install script is so large, we need several other programs to successfully get a machine provisioned
```
Comment (PromoFaux) [inline automated install/basic-install.sh:317]: These comments are from `development-v6`, one of the things that got messed up in the rebase. 

Might be easier to start the changes from fresh on a fresh pull of `development`

Comment (PromoFaux) [reply automated install/basic-install.sh:317]: Ref for the changed comments: https://github.com/pi-hole/pi-hole/pull/4084

But honestly, probably much cleaner just to start fresh. (Sorry for being a pain the last couple of days!!)

Comment (MichaIng) [reply automated install/basic-install.sh:317]: Ah yes that is true. I anyway need to rebase first and remove all the changed comments which are not intended as part of this PR.

Comment (MichaIng) [reply automated install/basic-install.sh]: Okay rebase done and conflicts resolved, including reverting the v6 comments 🙂.

**Comment by dschaper** [comment]
Does this still need hold status or can we clear it for 5.4?

**Comment by PromoFaux** [comment]
Have just updated it from latest dev, want to check it against the newer tests first....

**Review by PromoFaux — DISMISSED** [review]

**Review by PromoFaux — APPROVED** [review]
Sorry @MichaIng, I've mentally approved this so many times - and then just keep forgetting to!

**Pull request merged**